packages/sdk-codegen/src/sdkModels.ts

```diff
@@ -886,10 +920,65 @@ export class Method extends SchemadSymbol implements IMethod {
       this.addParam(api, body)
     }
     this.activityType = schema['x-looker-activity-type']
+    this.rateLimited = Method.isRateLimited(schema)
+  }
+
+  /**
+   * Check the many, many ways an endpoint can be flagged as rate-limited
+   *
+   * Because we LOVE standards. Yes we do.
+   *
+   * See https://github.com/OAI/OpenAPI-Specification/issues/1539 for more color commentary
+   *
+   * @param {OAS.OperationObject} op to check for rate limiting
+   * @returns {boolean} true if any rate limiting attribute is defined for the op
+   */
+  static isRateLimited(op: OAS.OperationObject): boolean {
+    if (op['x-looker-rate-limited']) return true
+
+    const many = [
+      'X-RateLimit-Limit',
+      'X-RateLimit-Remaining',
+      'X-RateLimit-Reset',
+      'X-Rate-Limit-Limit',
+      'X-Rate-Limit-Remaining',
+      'X-Rate-Limit-Reset',
+    ]
+    for (const flag of many) {
+      if (flag in op) return true
+    }
+
+    return false
+  }
+
+  mayUseRequestType(): boolean {
+    const [body] = this.bodyParams
+    /**
+     * if the body parameter is specified and is optional, at least 2 optional parameters are required
+     */
+    const offset = body && `required` in body && !body.required ? 1 : 0
+    return this.optionalParams.length - offset > 1
+  }
+
+  makeTypes(api: IApiModel): KeyList {
+    if (this.mayUseRequestType()) {
+      api.getRequestType(this)
```

Comment:
this reads a little funny to me. I've always felt that `getRequestType` **making** a request type if one doesn't exist is a little beyond the scope of the name. Reading this line the "command query separation" violation stands out to me even more. If breaking apart the getting and making is too cumbersomeI maybe we can settle for a kludgey name like `getOrCreateRequestType` ?

Reply:
Let's create a separate PR for this rename. The cross-package dependency issues with the rename are major until I get yarn workspace links working correctly